packages/@azure/servicebus/data-plane/test/testUtils.ts

```diff
@@ -40,13 +42,50 @@ export enum ClientType {
   TopicFilterTestSubscription
 }
 
-export function getSenderClient(
+export async function getSenderClient(
   namespace: Namespace,
   clientType: ClientType
-): QueueClient | TopicClient {
+): Promise<QueueClient | TopicClient> {
   switch (clientType) {
     case ClientType.PartitionedQueue:
-      return namespace.createQueueClient(process.env.QUEUE_NAME || "partitioned-queue");
+      const subscriptionId = process.env["AZURE_SUBSCRIPTION_ID"] || "";
+      const queueName = process.env.QUEUE_NAME || "partitioned-queue";
+      await msRestNodeAuth
+        .loginWithServicePrincipalSecret(
+          process.env.ARM_SERVICEBUS_CLIENT_ID || "",
```

Comment:
As this won't work unless you actually have credentials I think we should check that we have a non-empty value for these variables and remove the `|| ""` part from this code.

Reply:
Sure, @AlexGhiondea !